File: lab/ข้อ 11/-11.py
def input_num():
    n = int(input("enter your amount of number = "))
    number = []
    for i in range(n):
        na = int(input("enter your int number = "))
        number.append(na)
    print("your number is ",number)
    find_odd(n,number)
    find_even(n,number)
def find_odd(n,allna):
    odd = []
    posodd = []
    for i in range(n):
        if allna[i]%2!=0:
            k_num=allna[i]
            odd.append(k_num)
            posodd.append(i)
    output_odd(odd,posodd)
def find_even(n,allna):
    even = []
    poseven = []
    for i in range(n):
        if allna[i]%2==0:
            k_num = allna[i]
            even.append(k_num)
            poseven.append(i)
    output_even(even,poseven)
def output_odd(allodd,posodd):
    print("your odd number is",allodd)
    print("your odd position is",posodd)
    return 0
def output_even(alleven,poseven):
    print("your odd number is",alleven)
    print("your odd position is",poseven)
    return 0
# main
input_num()

File: lab/ข้อ 11/-11.cpp
#include "stdio.h"
#include "conio.h"
void inputarray(int size);
void find_odd(int n[], int allna);
void find_even(int n[], int allna);
void output_odd(int a[], int size);
void output_even(int a[], int size);
int main(){
	int n, i, array[n];
		printf("Input size of array : ");
		scanf("%d", &n);
	while (n < 1);
	printf("=======================================\n");
	inputarray(n);
	getch();
}
void Outputoddnumbers(int a[], int b[], int j)
{
	int i;
	printf("========== Oddnumber ========== \n");
	for (i = 0; i < j; i++)
	{
		printf("odd is %d \n", a[i]);
		printf("oddposition is %d \n", b[i]);
	}
	printf("Total oddnumber = %d \n",j);
}
void Outputevennumbers(int a[], int b[], int j)
{
	int i;
	printf("========= Evennumber ==========  \n");
	for (i = 0; i < j; i++)
	{
		printf("even is %d \n", a[i]);
		printf("evenposition is %d \n", b[i]);
	}
	printf("Total evennumber = %d \n\n",j);
}
void oddnumbers(int a[], int size)
{
	int i, j;
	j = 0;
	int odd[size], oddposition[size];
	for (i = 0; i < size; i++)
	{
		if (a[i] % 2 != 0)
		{
			odd[j] = a[i];
			oddposition[j] = i;
			j = j + 1;
		}
	}
	Outputoddnumbers(odd, oddposition, j);
}
void evennumbers(int a[], int size)
{
	int i, j;
	j = 0;
	int even[size], evenposition[size];
	for (i = 0; i < size; i++)
	{
		if (a[i] % 2 == 0)
		{
			even[j] = a[i];
			evenposition[j] = i;
			j = j + 1;
		}
	}
	Outputevennumbers(even, evenposition, j);
}
void inputarray(int size)
{
	int n = size;
	int i, array[n];
	for (i = 0; i < n; i++)
	{
		printf("Input element of array at [%d]: ", i);
		scanf("%d", &array[i]);
	}
	printf("\n\n");
	evennumbers(array, n);
	oddnumbers(array, n);
}
